### PR TITLE
Change default prometheus ports to new reserved Cilium ports

### PR DESCRIPTION
**Pull request by knfoo**
Cilium currently have 4 different prometheus exporters.
Cilium-agent, Cilium-operator, Cilium-proxy, and hubble. None of the
exporters ports have reserved ports in the prometheus docs and can
collision with other exporters by using the default ports.

The ports in this PR has been reserved: https://github.com/prometheus/prometheus/wiki/Default-port-allocations
Search look for Cilium and Hubble.

Signed-off-by: Karsten Nielsen <karsten.nielsen@ingka.ikea.com>

### CODE REVIEW

**Review by qmonnet — CHANGES_REQUESTED** [review]
Thanks for this work!

The CI caught a few things that would need to be addressed:

- Please add a commit object to your commit (you just have a description currently, the beginning of which is picked by GitHub as the commit title (“_Cilium currently have 4 different prometheus exporters._”), but this is not very descriptive of the patch).
- Please run `make -C install/kubernetes docs` and commit the changes.
- Please run `make -C Documentation update-cmdref` (I think?), and commit the changes.
- The [conformance test](https://github.com/cilium/cilium/runs/6830436885?check_suite_focus=true) is failing because there's still something trying to connect to port 9090 somewhere, but I haven't pinned it down yet :thinking:

**Review by joestringer — COMMENTED** [review]
Thanks for the submission. In addition to @qmonnet 's feedback above, could you also add a note to https://docs.cilium.io/en/latest/operations/upgrade/#upgrade-notes to describe that the default Prometheus ports were changed?

**Comment by knfoo** [comment]
@qmonnet not sure exactly what you mean with:

> Please add a commit object to your commit

Should I add a line above the commit description  ?
Should I do that with a `git commit --amend`or delete the pr and create a new one ?

**Comment by knfoo** [comment]
@qmonnet it seems that the Prometheus port is hardcoded in the conformance test: https://github.com/cilium/cilium/blob/afe5414737861eee1e84d48998d11d0397f22393/.github/workflows/tests-smoke.yaml#L191

I will change that as well to match the new default port.

**Comment by knfoo** [comment]
@joestringer when reading through this: https://docs.cilium.io/en/stable/contributing/development/contributing_guide/ it is a little unclear to me how I should do this.

Section 8 stats as you that I should put it in `Documentation/install/upgrade.rst` that seems to me to be version specific, so I am not sure where to put the release notes for this PR in that file.

Section 11 states that I can put in a release-note section in my commit message to override the release note text. Am I able to use the same formatting as the `upgrade.rst` file - more specific I am looking to include a table in the note ?

Can you point me in the right direction ?

**Comment by joestringer** [comment]
@knfoo Ah, it looks like the path was wrong. I opened #20184 to fix this up in the docs. The page I meant is `Documentation/operations/upgrade.rst`, you should see a "v1.12 upgrade notes" section under there where there should be an appropriate place to put the note. This is basically to allow folks to read that page only during upgrades and figure out if they should be aware of anything or make changes during the upgrade. This is different from the regular release notes which is much longer and includes things that are both relevant to upgrade but also just all general changes including features etc.

In this case for the release note itself, the PR title should be sufficient, so you do not need to worry about Step 11.

File: install/kubernetes/cilium/values.yaml
Comment (joestringer) [inline install/kubernetes/cilium/values.yaml:674]: After #20066 was merged, now these changes need to be made to `install/kubernetes/cilium/values.yaml.tmpl` instead, then run `make -C install/kubernetes` to update this file.

Comment (knfoo) [reply install/kubernetes/cilium/values.yaml:674]: I did a rebase and changed the `values.yaml.tmpl` file but I get this error:

```
make -C install/kubernetes
make: Entering directory '/home/kanie2/go/src/github.com/cilium/cilium/install/kubernetes'
: cilium/Chart.yaml
grep -lR -e 'version:' -e 'appVersion:' /home/kanie2/go/src/github.com/cilium/cilium/install/kubernetes/cilium/ \
        | xargs -L 1  sed -i -e 's/''\([vV]ersion:\) ''[0-9]\+\.[0-9]\+\.[0-9]\+.*''/\1 '1.11.90'/g'
sed -i 's;icon:.*;icon: https://cdn.jsdelivr.net/gh/cilium/cilium@master/Documentation/images/logo-solo.svg;' "/home/kanie2/go/src/github.com/cilium/cilium/install/kubernetes/cilium/Chart.yaml"
docker container run --rm --workdir /src/install/kubernetes --volume /home/kanie2/go/src/github.com/cilium/cilium/install/kubernetes/../..:/src --user "1000:1000" "docker.io/bmcustodio/helm-docs:"5ddabba"@sha256:"c4a91daac6bf9c181fac79e09712fcf1f9102fcab1bbacbc733883965f5fd244""
time="2022-06-14T19:41:04Z" level=info msg="Found Chart directories [cilium]"
time="2022-06-14T19:41:04Z" level=info msg="Generating README Documentation for chart /src/install/kubernetes/cilium"
: 
for crd in ciliumclusterwidenetworkpolicies ciliumnetworkpolicies ciliumidentities ciliumendpoints ciliumexternalworkloads ciliumegressgatewaypolicies ciliumnodes ciliumlocalredirectpolicies ciliumenvoyconfigs ciliumclusterwideenvoyconfigs ciliumbgploadbalancerippools ciliumbgppeeringpolicies ciliumendpointslices ciliumegressnatpolicies; do \
        grep -q $crd "/home/kanie2/go/src/github.com/cilium/cilium/install/kubernetes/cilium/Chart.yaml" \
        || (echo -e "$crd not found in "/home/kanie2/go/src/github.com/cilium/cilium/install/kubernetes/cilium/Chart.yaml".\nPlease update the chart to include $crd."; exit 1); \
done
helm lint --with-subcharts --values ./cilium/values.yaml ./cilium
==> Linting ./cilium
[ERROR] templates/: parse error at (cilium/templates/validate.yaml:38): unclosed action

Error: 1 chart(s) linted, 1 chart(s) failed
make: *** [Makefile:53: lint] Error 1
make: Leaving directory '/home/kanie2/go/src/github.com/cilium/cilium/install/kubernetes'
```
I am not sure what to do about that.

Comment (joestringer) [reply install/kubernetes/cilium/values.yaml:674]: :thinking: hmm, when I check out your PR and run that command it works. Which version of `helm` do you have installed locally?

Comment (joestringer) [reply install/kubernetes/cilium/values.yaml:674]: Worst case you can just skip the lint by doing `make -C install/kubernetes update-versions docs`.

Comment (knfoo) [reply install/kubernetes/cilium/values.yaml:674]: I have version: v3.5.4, I guess that is why and I should upgrade to a newer version.
I was just expecting the test to be run in the container as it was running in there ?

Comment (knfoo) [reply install/kubernetes/cilium/values.yaml:674]: I have done the steps and committed the changes. Should I resolve this conversation ?

Comment (joestringer) [reply install/kubernetes/cilium/values.yaml:674]: Only some of the commands in that step are run in the container today. I agree that running Helm in a container would be a reasonable way to mitigate this.

File: Documentation/operations/upgrade.rst
Comment (qmonnet) [inline Documentation/operations/upgrade.rst]:
```suggestion
* The default metrics exporter ports have changed to reserved ports in the
  `Prometheus project <https://github.com/prometheus/prometheus/wiki/Default-port-allocations>`__.
```

File: Documentation/operations/upgrade.rst
Comment (qmonnet) [inline Documentation/operations/upgrade.rst]:
```suggestion
  |                       | Cilium <= v1.11       | Cilium >= v1.12         |
```

Thanks for the table by the way, nice way to visualise the changes quickly.

**Review by qmonnet — CHANGES_REQUESTED** [review]
Minor nits on the docs, but the rest of the changes look all good to me, thanks for this work!

Can you please squash all your commits into a single one? All changes are related so I think it's fine to keep just one here. If you make sure to keep just the commit title and description from your first commit, this should also quiet checkpatch in the CI, who complains about some commit titles being too long or formatted on multiple lines (usually: one line for the title, 75 char max, then one blank line, then the commit description, then the `Signed-off-by:` tag at the end).

**Review by pchaigno — APPROVED** [review]
I reviewed CI changes and they look fine. Agree with Quentin that all changes should be squashed in a single commit.

**Review by qmonnet — APPROVED** [review]
Thanks!

File: Documentation/operations/upgrade.rst
Comment (joestringer) [reply Documentation/operations/upgrade.rst]: Looks like this change got missed in the latest update?

Comment (knfoo) [reply Documentation/operations/upgrade.rst]: :thinking:  Humm - I will have an other look at why

**Review by julianwiedmann — APPROVED** [review]

**Review by joestringer — APPROVED** [review]

**Comment by joestringer** [comment]
The test failure is unrelated to the code. With 4 reviews in I think this should be good enough to merge, no need to wait for each individual codeowner for a defaults change like this. Thanks!

**Pull request merged**
